Bug: Error while running kanata as a normal user `Failed to open the output uinput device`

### Requirements

- [x] I've searched [platform-specific issues](https://github.com/jtroo/kanata/blob/main/docs/platform-known-issues.adoc), [issues](https://github.com/jtroo/kanata/issues) and [discussions](https://github.com/jtroo/kanata/discussions) to see if this has been reported before.
- [x] My issue does not involve multiple simultaneous key presses, OR it does but I've confirmed it is not [key rollover or ghosting](https://github.com/jtroo/kanata/discussions/822).

### Describe the bug

I have followed the wiki at https://github.com/jtroo/kanata/wiki/Avoid-using-sudo-on-Linux but still cannot run kanata as a normal user on archlinux.

```
$ groups
me video input wheel uinput

$ cat /etc/udev/rules.d/99-keyboard.rules
KERNEL=="uinput", MODE="0660", GROUP="uinput", OPTIONS+="static_node=uinput"

$ cat kanata-minimal-config.kbd
;; minimal kanata config
(defcfg process-unmapped-keys yes)
(defsrc a b)
(deflayer foo b a)

$ kanata --debug -c kanata-minimal-config.kbd
11:03:02.1874 [INFO] kanata v1.9.0 starting
11:03:02.1878 [INFO] process unmapped keys: true
11:03:02.1880 [INFO] NOTE: kanata was compiled to never allow cmd
11:03:02.1880 [DEBUG] (1) kanata_parser::cfg::alloc: freeing allocations of length 0
11:03:02.1882 [INFO] config file is valid
11:03:02.1883 [ERROR] Failed to open the output uinput device. Make sure you've added the user executing kanata to the `uinput` group
11:03:02.1883 [DEBUG] (1) kanata_parser::cfg::alloc: freeing allocations of length 4
11:03:02.1883 [ERROR] Permission denied (os error 13)


Press enter to exit

Error: Permission denied (os error 13)
```

### Relevant kanata config

```text
;; minimal kanata config
(defcfg process-unmapped-keys yes)
(defsrc a b)
(deflayer foo b a)
```

### To Reproduce

Install kanata from AUR, or compile from source on arch. Create `kanata-minimal-config.kbd` as above, and run `kanata -c kanata-minimal-config.kbd`

### Expected behavior

kanata should start as usual.

### Kanata version

kanata 1.9.0

### Debug logs

```text
11:03:02.1874 [INFO] kanata v1.9.0 starting
11:03:02.1878 [INFO] process unmapped keys: true
11:03:02.1880 [INFO] NOTE: kanata was compiled to never allow cmd
11:03:02.1880 [DEBUG] (1) kanata_parser::cfg::alloc: freeing allocations of length 0
11:03:02.1882 [INFO] config file is valid
11:03:02.1883 [ERROR] Failed to open the output uinput device. Make sure you've added the user executing kanata to the `uinput` group
11:03:02.1883 [DEBUG] (1) kanata_parser::cfg::alloc: freeing allocations of length 4
11:03:02.1883 [ERROR] Permission denied (os error 13)
```

### Operating system and I/O mechanism

Arch Linux

### Additional context

kanata runs using sudo as expected.

--- a/docs/setup-linux.md
+++ b/docs/setup-linux.md
@@ -5,7 +5,17 @@ In Linux, kanata needs to be able to access the input and uinput subsystem to in
 ### 1. If the uinput group does not exist, create a new group
 
 ```bash
-sudo groupadd uinput
+sudo groupadd --system uinput
+```
+
+If the `uinput` group already exists,
+it may have been created without the `--system` flag.
+First delete it then re-add again with the above command.
+
+To delete:
+
+```
+sudo groupdel uinput
 ```
 
 ### 2. Add your user to the input and the uinput group

--- a/src/kanata/mod.rs
+++ b/src/kanata/mod.rs
@@ -326,6 +326,8 @@ use once_cell::sync::Lazy;
 static MAPPED_KEYS: Lazy<Mutex<cfg::MappedKeys>> =
     Lazy::new(|| Mutex::new(cfg::MappedKeys::default()));
 
+const LINUX_PERMISSIONS_ERROR: &str = "Failed to open the output uinput device. Make sure you added the user executing kanata to the 'uinput' group and that the 'uinput' group is configured correctly.\nSee for more detail: https://github.com/jtroo/kanata/blob/main/docs/setup-linux.md";
+
 impl Kanata {
     pub fn new(args: &ValidatedArgs) -> Result<Self> {
         let cfg = match cfg::new_from_file(&args.paths[0]) {
@@ -351,9 +353,7 @@ impl Kanata {
         ) {
             Ok(kbd_out) => kbd_out,
             Err(err) => {
-                error!(
-                    "Failed to open the output uinput device. Make sure you've added the user executing kanata to the `uinput` group"
-                );
+                error!("{LINUX_PERMISSIONS_ERROR}");
                 bail!(err)
             }
         };
@@ -521,9 +521,7 @@ impl Kanata {
         ) {
             Ok(kbd_out) => kbd_out,
             Err(err) => {
-                error!(
-                    "Failed to open the output uinput device. Make sure you've added the user executing kanata to the `uinput` group"
-                );
+                error!("{LINUX_PERMISSIONS_ERROR}");
                 bail!(err)
             }
         };

